flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -313,6 +327,19 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
 			throw new DatabaseNotExistException(getName(), tablePath.getDatabaseName());
 		}
 
+		if (table instanceof ConnectorCatalogTable) {
+
+			if (tableExists(tablePath)) {
```

Comment:
I think we'll need some synchronization here if we want to achieve thread safety.